### PR TITLE
soroban-rpc: Auth Next

### PR DESCRIPTION
**Pull request by paulbellamy**
Staging area for soroban-rpc auth-next changes

### CODE REVIEW

**Review by tsachiherman — COMMENTED** [review]
The change here looks very reasonable. One thing that I think we should work on is decoupling the C calls from the handler.
i.e. all the C calls should be placed in it's own package, exposing only GO datatypes. That would allow us to decouple the handler logic from the C wrapper code.

File: cmd/soroban-rpc/internal/methods/simulate_transaction.go [hunk 1]
```diff
@@ -171,10 +172,11 @@ func NewSimulateTransactionHandler(logger *log.Entry, networkPassphrase string,
 			}
 		}
 		li := C.CLedgerInfo{
-			network_passphrase: C.CString(networkPassphrase),
-			sequence_number:    C.uint(latestLedger),
-			protocol_version:   20,
-			timestamp:          C.uint64_t(time.Now().Unix()),
+			// TODO: Read network_id, 32 bytes
```
Comment (sreuland) [inline cmd/soroban-rpc/internal/methods/simulate_transaction.go:175]: can this TODO be resolved now, not sure what the blocker is?

File: cmd/soroban-rpc/internal/methods/simulate_transaction.go [hunk 2]
```diff
@@ -98,13 +98,14 @@ type SimulateTransactionCost struct {
 }
 
 type InvokeHostFunctionResult struct {
-	XDR string `json:"xdr"`
+	XDR       string   `json:"xdr"`
```
Comment (tsachiherman) [inline cmd/soroban-rpc/internal/methods/simulate_transaction.go:101]: Could we rename this field to something a bit more descriptive ?
i.e. "output", "returnValue", "outcome"

**Comment by 2opremio** [comment]
@paulbellamy should we close this?

**Comment by paulbellamy** [comment]
Closed in favour of `auth-next2` branch

**Pull request closed without merging**